### ISSUE TITLE
Ring Video Doorbell Pro 2

### ISSUE THREAD

**Issue by Marfre888** [dgreif/ring#602]
# Bug Report

### Describe the Bug

So I've just installed the Ring Video Doorbell Pro 2, and it is just impressive the difference between the original Pro. I never thought the improvement would be this big. Alas, there are obviously some problems which have come up with the new doorbell.

I'm happy to report that I am able to get snapshots from doorbell inside of homekit. The problems start when you click on the tile. The video is incredibly laggy, around 1-2frames/s (in the ring app it's fine). I also get no sound or static, whereas in the Ring app, the doorbell background ambient sound is significantly quieter (a welcome improvement).

Please let me know how I am able to help you get the fix out.

Thanks

### Data Discovery

[ring_discovery_videodoorbellpro2.txt](https://github.com/dgreif/ring/files/6244659/ring_discovery_videodoorbellpro2.txt)

### Environment

 - OS: Raspbian
 - Node.js: 14.16.0
 - NPM: 6.14.11
 - homebridge-ring: 9.15.5
 - homebridge: 1.3.4

**Comment by dgreif** [comment]
Glad to hear you like the Pro 2!  Honestly, I think the only way for me to debug the issues you are describing will be for me to buy one myself.  I'm planning to buy more cameras soon, but was hoping to wait until Amazon Prime Day to get good deals.  If anyone else is having similar issues, feel free to throw a comment below.

**Comment by Marfre888** [comment]
> Glad to hear you like the Pro 2! Honestly, I think the only way for me to debug the issues you are describing will be for me to buy one myself. I'm planning to buy more cameras soon, but was hoping to wait until Amazon Prime Day to get good deals. If anyone else is having similar issues, feel free to throw a comment below.

Oh that sucks.. what kind of data would you need? Would you be able to get the data you need if I gave you access to my doorbell?

**Comment by dgreif** [comment]
Yeah, I could get what I need if you give me access to just that camera.  If you are comfortable with that, you can share access with `dustin.greif@gmail.com`. I haven't had a lot of free time, so no promises on when I'll be able to get to it

**Comment by Marfre888** [comment]
So this morning I opened up the camera tile, and the sound started working, but it was a lot more staticky than in the ring app. (I didn't test the mic). Then I periodically tried to open it again over the course of a couple of minutes, and it would randomly give me sound but for the most part, I just got silence.

Video stream is still noticeably jittier (around 4fps). Could it simply be that Ring has changed the way they are compressive audio/video feeds. Since audio is now advertised as 'HD Audio' and 'HD Video'.

If you want to take a look I've shared to doorbell with you. Let me know when you're finished playing.

**Comment by Marfre888** [comment]
Hi @dgreif 

Have you had any chance to look at this by any chance?

Could it be that the new Pro 2 runs at 24f not 10f? And that’s the reason why it’s not working properly

**Comment by dgreif** [comment]
@Marfre888 I haven't tried it out yet because I was working on some changes to how streams connect that I hoped would fix it.  However, after looking at the specs, I think your idea is probably correct.  The new Pro 2 is a much higher resolution so the stream is sending a _lot_ more data.  It's possible that HomeKit just isn't handing it well, causing the poor performance you are now seeing.  It _may_ be possible to request a lower quality stream from the camera, but that will take some time for me to investigate.  I haven't accepted you sharing offer yet, but if you don't mind keeping that out there, I'll dig into it when I get a chance.  Probably won't be for a while though.  Thanks for being patient!

**Comment by Marfre888** [comment]
Ahh sure.. I’ll keep the sharing open for you, I just wasn’t sure you realised I sent it to you.

Snapshot loads fine that clearly hasn’t changed. HDR on/off makes no difference. Audio seems to work about 3/10 times, but it is significantly noisier than on the Ring app. Video loads fine but but is very stuttery

Looking forward for your fix. Thanks

**Comment by eugenesoo** [comment]
i wanted to add that the motion sensor for the ring doorbell pro 2 doesn’t seem to be working, @Marfre888 did you have this issue as well?

**Comment by dgreif** [comment]
@Marfre888 are you still experiencing issues with streaming from the doorbell pro 2?

**Comment by Marfre888** [comment]
> Ok, I think it's probably as good as it's going to get. Sorry I can't improve it more, but the Ring app is probably going to be your best experience. Thanks for being patient on this!

Hi @dgreif,

I'm not sure why you think it can't be improved further.. before the Pro 2 I had the OG Pro, this worked wonderfully in Homekit, with no sound issues or laggy video.

With the Pro 2, the video is significantly laggier than the OG Pro and I get random audio static. It would be great if what is being pushed to Homekit is the same quality as what I was getting with the OG Pro

**Comment by dgreif** [comment]
It all had to do with how Ring is encoding the stream, which I don't have a lot of control over. They must do different encoding for the pro 2

**Comment by Marfre888** [comment]
> It all had to do with how Ring is encoding the stream, which I don't have a lot of control over. They must do different encoding for the pro 2

@dgreif I understand that, however on the Ring website preview loads fine and there is no stutter or sound issues.

Wouldn't that point to an issue with how ffmpeg is handling the stream it is receiving? My invite is still extended to you if you want to add the camera into your setup to have a look..

**Comment by dgreif** [comment]
> however on the Ring website preview loads fine and there is no stutter or sound issues.

Can you clarify what you are looking at on ring.com?  Is it a live view or just recordings?

**Comment by Marfre888** [comment]
Live view

**Comment by dgreif** [comment]
Whoah this is a new feature! Will need to spend some time figuring out how they are streaming it.

**Comment by Marfre888** [comment]
> Whoah this is a new feature! Will need to spend some time figuring out how they are streaming it.

Haha.. is it? I could have sworn I had it for the past year :)

**Comment by dgreif** [comment]
Well it's new to my eyes!  Having this in the browser is extremely helpful for seeing how the APIs work, and actually includes a different API then we currently use today. I'll be digging into this and see if I can get it to work from the plugin. Once it's live, we'll see if your camera works any better.
